src/components/CommentCards.jsx

```diff
@@ -37,15 +44,42 @@ export const CommentCards = ({ reviewId }) => {
         let realDate = `${date[2]}/${date[1]}/${date[0]}`;
         return (
           <li key={comment.comment_id}>
-            
             <div id="comment-sub-info">
               <h4 className=" sub-info emphasise">@{comment.author}</h4>
               <p className=" sub-info">{realDate}</p>
             </div>
             <p className="comment">"{comment.body}"</p>
 
             <CommentVotes votes={comment.votes} />
-            <Divider id='divider'/>
+            {isLoggedIn && user.username === comment.author ? (
+              <Button
+                disabled={isDeleting}
+                variant="dark button"
+                className="Read-More"
+                onClick={() => {
+                  setIsDeleting(true);
+                  deleteComment(comment.comment_id).then(setIsDeleting(false));
```

Comment:
`setIsDeleting` is being invoked before the `deleteComment` promise resolves here - refactor to `() => setIsDeleting(false)`